lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -401,6 +409,30 @@ public function getMaxResults()
         return $this->maxResults;
     }
 
+    /**
+     * Gets the current lock mode  for this query
+     *
+     * @return int|null
+     */
+    public function getLockMode()
+    {
+        return $this->lockMode;
+    }
+
+    /**
+     * Set lock mode use one of the constants from LockMode::class locks are only added to SELECT queries
+     *
+     * Optimistic locking *is not* supported by DBAL.
+     *
+     * @param int $lockMode
+     */
+    public function setLockMode(int $lockMode)
+    {
+        $this->lockMode = $lockMode;
```

Comment:
The ORM query builder includes defensive coding around not allowing locks to be set outside of a transaction (they would be immediately released), should that be added here? 

I lean towards no given the warning comments at the top of the class that it is the user's responsibility to use this class to generate valid queries.